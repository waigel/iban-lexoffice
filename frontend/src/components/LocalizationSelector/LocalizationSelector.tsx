import {useTolgee, useTranslate} from '@tolgee/react';
import React from 'react';

export const LocalizationSelector = () => {
  const {t} = useTranslate();
  const tolgee = useTolgee(['language']);

  const localizations = [
    {label: t('de-DE'), value: 'de-DE'},
    {label: t('en-US'), value: 'en-US'},
  ];

  return (
    <>
      <select
        onChange={e => tolgee.changeLanguage(e.target.value)}
        value={tolgee.getLanguage()}
        className="bg-transparent ring-0"
      >
        {localizations.map(localization => (
          <option
            className="bg-black ring-0"
            key={localization.value}
            value={localization.value}
          >
            {localization.label}
          </option>
        ))}
      </select>
    </>
  );
};
