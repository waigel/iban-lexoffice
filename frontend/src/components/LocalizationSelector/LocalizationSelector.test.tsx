import React from 'react';
import TestRenderer, {act} from 'react-test-renderer';
import {LocalizationSelector} from './LocalizationSelector';

jest.mock('@tolgee/react', () => ({
  useTranslate: () => ({t: jest.fn(key => key), isLoading: false}),
  useTolgee: () => ({
    isLoading: false,
    getLanguage: jest.fn(),
    changeLanguage: jest.fn(),
  }),
}));

it('should test the LocalizationSelector component is rendered correctly', () => {
  const tree = TestRenderer.create(<LocalizationSelector />);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should test the LocalizationSelector can change language', () => {
  const tree = TestRenderer.create(<LocalizationSelector />);
  act(() => {
    tree.root.findByType('select').props.onChange({target: {value: 'test'}});
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
