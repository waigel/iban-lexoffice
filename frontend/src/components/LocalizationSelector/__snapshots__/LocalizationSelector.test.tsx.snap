// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the LocalizationSelector can change language 1`] = `
<select
  className="bg-transparent ring-0"
  onChange={[Function]}
>
  <option
    className="bg-black ring-0"
    value="de-DE"
  >
    de-DE
  </option>
  <option
    className="bg-black ring-0"
    value="en-US"
  >
    en-US
  </option>
</select>
`;

exports[`should test the LocalizationSelector component is rendered correctly 1`] = `
<select
  className="bg-transparent ring-0"
  onChange={[Function]}
>
  <option
    className="bg-black ring-0"
    value="de-DE"
  >
    de-DE
  </option>
  <option
    className="bg-black ring-0"
    value="en-US"
  >
    en-US
  </option>
</select>
`;
